src/crafting.cpp

```diff
@@ -1293,6 +1307,20 @@ void pick_recipes(const inventory &crafting_inv,
                         continue;
                     }
                 }
+                if(search_qualities) {
+                  bool match_found = false;
+                  itype *it = item::find_type(rec->result);
+                  for(std::map<std::string, int>::const_iterator quality = it->qualities.begin(); quality != it->qualities.end() && !match_found; ++quality) {
```

Comment:
Instead of writing the iterator type yourself, you could use `auto` and let the compiler figure it out.

It could also be done with a (equivalent) range-based loop:
```C++
for( auto & e : it->qualities ) {
   quality::get_name( e.first );
   ...
```